graylog2-web-interface/src/components/inputs/CreateInputControl.jsx

```diff
@@ -54,6 +56,14 @@ const CreateInputControl = createReactClass({
 
   _openModal(event) {
     event.preventDefault();
+    const { selectedInput } = this.state;
+
+    const customConfiguration = PluginStore.exports('inputConfiguration').filter(inputConfig => inputConfig.matches(selectedInput));
```

Comment:
As far as I know by the usual unwritten convention, the plugin provides a `type` static field and the web interface does the matching. I think that's a bit easier for plugin authors (they don't have to write code to do the matching themselves), and that way we also avoid duplicating matches in different places. That is, unless there's some particular reason to do it in a different way here.

Reply:
The idea was to allow a matcher, so we can match the beginning of the input's type string and use it for every input starting with `org.graylog.integrations.aws`. If we want to do that, we should also allow specifying a `type` key which the implementation falls back to if `matches` is not defined.

@kyleknighted: Do we really need to match a substring? Afaics we only have a single input class named `org.graylog.integrations.aws.inputs.AWSInput`, so we could also use a constant string comparison here?